pkg/morph/event/listener.go

```diff
@@ -313,6 +345,15 @@ func (s listener) Stop() {
 	s.subscriber.Close()
 }
 
+func (s listener) RegisterBlockHandler(handler BlockHandler) {
```

Comment:
Seems like missing block height there. And also `blockHandlers map[uint32]BlockHandler`.

Reply:
`RegisterBlockHandler` is a general purpose method. I will add the ability to subscribe to particular block height later. There will be something like this functionality:
```
// EqualHeightBlockHandler returns BlockHandler that calls bHandler only for blocks with
// Index == height.
func EqualHeightBlockHandler(height uint32, bHandler BlockHandler) BlockHandler {
	return func(b *block.Block) {
		if b.Index != height {
			return
		}

		bHandler(b)
	}
}
```